### PR TITLE
Merging test automation branch into dev to avoid future conflicts

### PR DESCRIPTION
**Pull request by NithinKuruba**

### CODE REVIEW

File: e2e/cypress/support/auth-commands.ts [hunk 1]
```diff
@@ -116,6 +123,18 @@ Cypress.Commands.add('getAccessToken', (client_id: string, client_secret: string
   cy.log('> Get Token')
 })
 
+Cypress.Commands.add('getServiceOrRouteID', (configType: string) => {
+  const config = configType.toLowerCase()
+  debugger
```
Comment (Jonesy) [inline e2e/cypress/support/auth-commands.ts:128]: Should this stray debugger be removed?

Comment (NithinKuruba) [reply e2e/cypress/support/auth-commands.ts]: Removed

File: src/nextapp/components/forms/list-input.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ interface ListInputProps {
   placeholder?: string;
   required?: boolean;
   value: string;
+  dataTestId?: string;
```
[the remark this answers is not in the document]
Comment (NithinKuruba) [reply src/nextapp/components/forms/list-input.tsx]: yep should be good idea.

File: src/nextapp/pages/devportal/requests/new/[id].tsx [hunk 3]
```diff
@@ -271,7 +272,12 @@ const NewRequestsPage: React.FC<
                   icon: FaBook,
                   label: (
                     <Box>
-                      <Text fontWeight="bold">{e.name}</Text>
+                      <Text
+                        fontWeight="bold"
+                        data-testid={'access-rqst-app-env-' + e.name}
```
[the remark this answers is not in the document]
Comment (NithinKuruba) [reply src/nextapp/pages/devportal/requests/new/[id].tsx]: done

File: src/nextapp/components/products-list/add-environment.tsx [hunk 4]
```diff
@@ -55,7 +57,7 @@ const AddEnvironment: React.FC<AddEnvironmentProps> = ({
 
   return (
     <Menu>
-      <MenuButton as={Button} variant="unstyled" data-testid="prd-env-add-btn">
+      <MenuButton as={Button} variant="unstyled" data-testid={productName.toLowerCase().replaceAll(' ', '-') + "-add-env-btn"}>
```
[the remark this answers is not in the document]
Comment (NithinKuruba) [reply src/nextapp/components/products-list/add-environment.tsx]: done

File: src/nextapp/components/products-list/add-environment.tsx [hunk 5]
```diff
@@ -66,7 +73,7 @@ const AddEnvironment: React.FC<AddEnvironmentProps> = ({
               key={e.value}
               onClick={onSelect(e.value)}
               value={e.value}
-              data-testid={'prd-env-item-' + e.value}
+              data-testid={kebabCase(productName) + `-prd-env-item-${e.value}`}
```
Comment (Jonesy) [inline src/nextapp/components/products-list/add-environment.tsx:76]: Prefer ```data-testid={`${kebabCase(productName)}-prd-env-item-${e.value}`}``` for consistency

File: src/nextapp/components/products-list/add-environment.tsx [hunk 6]
```diff
@@ -10,6 +10,7 @@ import {
 import { useQueryClient } from 'react-query';
 import { ADD_ENVIRONMENT } from '@/shared/queries/products-queries';
 import { useApiMutation } from '@/shared/services/api';
+import { kebabCase } from 'lodash';
```
Comment (Jonesy) [inline src/nextapp/components/products-list/add-environment.tsx:13]: See first comment for correct import declaration

File: src/nextapp/components/environments-list/environments-list.tsx [hunk 7]
```diff
@@ -122,7 +127,7 @@ const EnvironmentsList: React.FC<EnvironmentsListProps> = ({ data }) => {
                 <Button
                   size="xs"
                   variant="outline"
-                  data-testid="prd-env-edit-btn"
+                  data-testid={kebabCase(product.name) + `-${e.name}-edit-btn`}
```
Comment (Jonesy) [inline src/nextapp/components/environments-list/environments-list.tsx:130]: Duplicate of earlier revision request

**Review by Jonesy — CHANGES_REQUESTED** [review]

**Review by Jonesy — APPROVED** [review]

**Pull request merged**